### PR TITLE
Set availability for mentors

### LINKED ISSUE #118: Allow mentor to set their availability to false
We want to allow mentors to set their availability, for that we are going to add a new `available` property to the user model.

1. Allow to mentor to set that prop as `true` or `false`. We are going to reuse our existing API to edit the user model.
For new mentor, that prop will be `true` by default.
2. Write (and run) a script that migrate all existing mentors and add this prop as `true`.

Request
```
PUT /users/{user._id}

{
  available: true | false
}
```

Response
```
{ success: true }
```

**Restrictions**
- Only current user can update this value
- Admin can update this value for any other user
- Only users with `Mentor` role can update this value

### PR DESCRIPTION
**Pull request by crysfel**
As a mentor I want to set my availability, so I can hide my profile from the mentor search results.

closes #118

### CODE REVIEW

**Comment by brentmclark** [comment]
Code looks good to me, but I can't shake the feeling that excluding unavailable mentors from the mentors list is cutting too deep.  Maybe in the UI you could continue to show the unavailable mentors, but replace the connections section with some copy that says, for example, `this mentor has marked themselves unavailable`.

File: scripts/set-availability.js [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+ *  $ yarn user:availability
+ * 
+ */
+const mongoose = require('mongoose');
+const dotenv = require('dotenv');
+
+dotenv.config();
+
+(async () => {
+  mongoose.connect(process.env.MONGO_DATABASE_URL, { useNewUrlParser: true });
```
Comment (arku) [inline scripts/set-availability.js]: @crysfel The [mongoose.connect](https://mongoosejs.com/docs/api.html#mongoose_Mongoose-connect) function returns a Promise. Shouldn't we await a successful connection? 🤔 

```
await mongoose.connect(process.env.MONGO_DATABASE_URL, { useNewUrlParser: true });
```

/cc @brentmclark

File: scripts/set-availability.js [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+
+
+  const User = mongoose.model('User', {
+    _id: String,
+    available: Boolean,
+    email: String,
+    roles: Array,
+  });
+
+  const users = await User.find({ roles: 'Mentor', available: undefined }).exec();
```
Comment (arku) [inline scripts/set-availability.js]: Looks like we are only interested in the number of mentors without availability set. What do you think of using [`count`](https://mongoosejs.com/docs/api.html#model_Model.count) here?

```js
const mentorCount = await User.count({ roles: 'Mentor', available: undefined });
```

File: scripts/set-availability.js (hunk 1 above)
Comment (brentmclark) [reply scripts/set-availability.js]: Seems reasonable to me.  Avoids a race condition.

Comment (crysfel) [reply scripts/set-availability.js]: Good catch! 👌

File: scripts/set-availability.js (hunk 2 above)
Comment (crysfel) [reply scripts/set-availability.js]: Makes sense!!

**Comment by crysfel** [comment]
@brentmclark @arku Thank you so much for your review! I've implemented the updates

**Comment by crysfel** [comment]
> Code looks good to me, but I can't shake the feeling that excluding unavailable mentors from the mentors list is cutting too deep. Maybe in the UI you could continue to show the unavailable mentors, but replace the connections section with some copy that says, for example, `this mentor has marked themselves unavailable`.

Yes! I like this idea, I will update the code to return the mentors but with `available=false` so the UI can take those out

**Pull request merged**